services/src/main/java/org/keycloak/theme/beans/LocaleBean.java

```diff
@@ -73,6 +72,18 @@ public boolean isRtl() {
         return rtl;
     }
 
+    public boolean isRtl(String languageTag) {
+        if (CACHED_RTL_LANGUAGE_CODES.containsKey(languageTag)) {
+            return CACHED_RTL_LANGUAGE_CODES.get(languageTag);
+        }
+
+        boolean isRTL = new ULocale(languageTag).isRightToLeft();
+
+        CACHED_RTL_LANGUAGE_CODES.put(languageTag, isRTL);
+
+        return isRTL;
```

Comment:
Try to leverage modern APIs like `HashMap#computeIfAbsent` here to reduce boilerplate code and avoid concurrency issues.